### PR TITLE
Finish MT9M114 driver

### PR DESCRIPTION
**Pull request by kwagyeman**
Everything works except:
* Need to update bayer image processing code to handle all possible pattern options versus 1.
* Set_contrast appears to do nothing. Verified it's setting the right register however per the datasheet.
* Set_windowing doesn't really work. You can get cropping and a frame rate boost if you shrink the vertical res only. Need to work on the code more.
* Bigger issue with the above... changing the resolution/windowing wipes exposure settings making the 100 FPS example not work. Not really sure if this can be fixed given the way the sensor works.

### CODE REVIEW

File: src/omv/common/sensor.h [hunk 1]
```diff
@@ -38,12 +38,12 @@
 #define OV7725_ID           (0x77)
 #define OV9650_ID           (0x96)
 #define MT9V034_ID          (0x13)
-#define MT9M114_ID          (0x81)
```
Comment (iabdalkader) [inline src/omv/common/sensor.h:41]: Let's keep it here to match the order of init.

**Comment by iabdalkader** [comment]
>     * Bigger issue with the above... changing the resolution/windowing wipes exposure settings making the 100 FPS example not work. Not really sure if this can be fixed given the way the sensor works.

That's easy we can maintain the reg values in some structure, and write again after res changes or whenever it's needed.

File: src/omv/common/sensor.h (hunk 1 above)
Comment (kwagyeman) [reply src/omv/common/sensor.h:41]: Fixed

**Comment by kwagyeman** [comment]
It's ready

**Pull request merged**